sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -41,6 +43,11 @@ private[spark] object CosmosClientCache {
           .key(cosmosClientConfiguration.key)
           .endpoint(cosmosClientConfiguration.endpoint)
           .userAgentSuffix(cosmosClientConfiguration.applicationName)
+          .throttlingRetryOptions(
```

Comment:
This is to avoid 429s surfacing? Should this be an opt-in config? Otherwise users might not get signals of why their job is taking longer?

Reply:
Metrics - with RU budget otherwise ingestion jobs would be bound to fail all the time. Even in current OLTP connector endless retries on 429 are done.

Reply:
Otherwise 429 would result in hard failures of the spark job - and especially in batches that would potentially mean reprocessing hours of work